pkg/upload/js.go

```diff
@@ -16,9 +24,163 @@ type JsOptions struct {
 	Path        utils.Paths `arg:"" name:"path" help:"Path to a directory of source maps and bundles to upload" type:"path" default:"."`
 }
 
+// Resolve the project if it isn't specified using the currend working directory
+func resolveProjectRoot(projectRoot string, path string) string {
+	if projectRoot != "" {
+		return projectRoot
+	}
+	workingDirectory, err := os.Getwd()
+	if err != nil {
+		return path
+	}
+	return workingDirectory
+}
+
+// Attempt to parse information from the package.json file if values aren't provided on the command line
+func ResolveVersion(versionName string, projectRoot string, logger log.Logger) (string, error) {
+	if versionName != "" {
+		return versionName, nil
+	}
+	checkPath, err := filepath.Abs(projectRoot)
+	if err != nil {
+		return "", fmt.Errorf("unable to make project root an absolute path %s", projectRoot)
+	}
+	// Walk up the folder structure as far as possible
+	for filepath.Dir(checkPath) != checkPath {
+		packageJson := filepath.Join(checkPath, "package.json")
+		if !utils.FileExists(packageJson) {
+			checkPath = filepath.Dir(checkPath)
+			continue
+		}
+		file, err := os.Open(packageJson)
+		if err != nil {
+			logger.Warn(fmt.Sprintf("unable to open %s: %s", packageJson, err))
+			return "", nil
+		}
+		byteValue, err := io.ReadAll(file)
+		if err != nil {
+			logger.Warn(fmt.Sprintf("unable to read %s: %s", packageJson, err))
+			return "", nil
+		}
+		var parsedPackageJson map[string]interface{}
+		err = json.Unmarshal(byteValue, &parsedPackageJson)
+		if err != nil {
+			logger.Warn(fmt.Sprintf("unable to parse %s: %s", packageJson, err))
+			return "", nil
+		}
+		if parsedPackageJson["version"] == nil {
+			logger.Warn(fmt.Sprintf("missing version in %s", packageJson))
```

Comment:
This feels like more of a Debug message to say "no version found in ..."

Reply:
A package.json file must contain a version - https://docs.npmjs.com/creating-a-package-json-file#packagejson-fields.